File: app/src/main/java/com/example/guessthenumber/MainActivity.java
package com.example.guessthenumber;

import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    int result;
    static int getRandomNumber( int max, int min)
    {
        return ( int )((Math.random()
                * (max - min)) + min);
    }
    public void makeToast(String str)
    {
        Toast.makeText(
                        MainActivity. this ,
                        str,
                        Toast.LENGTH_SHORT)
                .show();
    }
    public void clickFunction(View view)
    {
        int userGuessing;
        EditText variable = findViewById(R.id.editId);
        userGuessing = Integer.parseInt(
                variable
                        .getText()
                        .toString());
        if (userGuessing < result) {
            makeToast("Предложенное число больше. Попробуйте снова!");
        }
        else if (userGuessing > result) {
            makeToast("Предложенное число меньше. Попробуйте снова!");
        }
        else {
            makeToast(
                    "Поздравляем! Вы угадали число!" );
        }
    }
    @Override
    protected void onCreate(
            Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        int min = 1 ;
        int max = 100 ;
        result = getRandomNumber(min, max);
    }
}

File: app/src/main/java/com/example/guessthenumber/MainActivity.kt
package com.example.guessthenumber

import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var res = 0
    private fun makeToast(str: String?) {
        Toast.makeText(
            this@MainActivity,
            str,
            Toast.LENGTH_SHORT
        )
            .show()
    }

    fun clickFunction(view: View) {
        val userGuessing: Int
        val variable = findViewById<EditText>(R.id.editId)
        userGuessing =
            variable
                .text
                .toString().toInt()
        if (userGuessing < res) {
            makeToast("Предложенное число больше. Попробуйте снова!")
        } else if (userGuessing > res) {
            makeToast("Предложенное число меньше. Попробуйте снова!")
        } else {
            makeToast(
                "Поздравляем! Вы угадали число!"
            )
        }
    }

    override fun onCreate(
        savedInstanceState: Bundle?
    ) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val min = 1
        val max = 100
        res = getRandomNumber(min, max)
    }

    companion object {
        fun getRandomNumber(max: Int, min: Int): Int {
            return ((Math.random() * (max - min)) + min).toInt()
        }
    }

    fun click2(view: View) {
        setContentView(R.layout.activity_about)
    }
}
